File: GXpert/GXpert.Web/Modules/Institute/InstituteStudent/InstituteStudentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.InstituteStudent")]
[BasedOnRow(typeof(InstituteStudentRow), CheckNames = true)]
public class InstituteStudentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string StudentPrn { get; set; }
    public string InstituteName { get; set; }
    [QuickFilter]
    public string ClassTitle { get; set; }
    [EditLink]
    public string Division { get; set; }
    public string RollNumber { get; set; }
    public string Description { get; set; }
    public string AcademicYearName { get; set; }
    public string Semester { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }

}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteStudentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { InstituteStudentRow } from "./InstituteStudentRow";

export interface InstituteStudentColumns {
    Id: Column<InstituteStudentRow>;
    StudentPrn: Column<InstituteStudentRow>;
    InstituteName: Column<InstituteStudentRow>;
    ClassTitle: Column<InstituteStudentRow>;
    Division: Column<InstituteStudentRow>;
    RollNumber: Column<InstituteStudentRow>;
    Description: Column<InstituteStudentRow>;
    AcademicYearName: Column<InstituteStudentRow>;
    Semester: Column<InstituteStudentRow>;
    StartDate: Column<InstituteStudentRow>;
    EndDate: Column<InstituteStudentRow>;
}

export class InstituteStudentColumns extends ColumnsBase<InstituteStudentRow> {
    static readonly columnsKey = 'Institute.InstituteStudent';
    static readonly Fields = fieldsProxy<InstituteStudentColumns>();
}